homeassistant/components/proxmoxve/__init__.py

```diff
@@ -156,7 +156,7 @@ def build_client() -> ProxmoxAPI:
 
             for vm_id in node_config["vms"]:
                 coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, vm_id, TYPE_VM
+                    hass, proxmox, host_name, node_name, vm_id, ProxmoxType.QEMU
```

Comment:
To be more future-proof - you should create a new `coordinator.py` module with separate classes for QEMU and LXC.
For now they both return `ProxmoxVMData` but later you can use TypeVar to extend that.

```python
class ProxmodCoordinator(DataUpdateCoordinator[ProxmoxVMData]):
    ...
class ProxmodQemuCoordinator(ProxmodCoordinator):
    ...
class ProxmodLxcCoordinator(ProxmodCoordinator):
    ...
```

```python
for vm_id in node_config["vms"]:
    coordinator = ProxmodQemuCoordinator(hass, proxmox, host_name, node_name, vm_id)

for container_id in node_config["containers"]:
    coordinator = ProxmodLxcCoordinator(hass, proxmox, host_name, node_name, container_id)
```

Reply:
Included in [d871371](https://github.com/home-assistant/core/pull/90371/commits/d87137105480f57b6c9f6b8f61718347dfe07d7c)